scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+from neo4j.v1 import GraphDatabase
+
+
+# In[3]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[4]:
+
+REACTOMEPATHWAY = 'PC7_7459'  # "Signaling by Hippo" in Reactome
+GOPATHWAY_A = 'GO:0035329'    # "hippo signaling" in the Gene Ontology
+GOPATHWAY_B = 'GO:0035330'    # "regulation of hippo signaling" in the Gene Ontology
+
+
+# In[5]:
+
+def pathway_query(path_id, node_type='BiologicalProcess', particip='GpBP'):
+    query = '''
+    MATCH (node:{})-[:PARTICIPATES_{}]-(gene)
+    WHERE node.identifier = "{}"
+    RETURN
+      gene.identifier as entrez_gene_id,
+      gene.name as gene_symbol
+    ORDER BY gene_symbol
+    '''.format(node_type, particip, path_id)
+    return query
+
+
+# In[6]:
+
+query_r = pathway_query(REACTOMEPATHWAY, node_type='Pathway', particip='GpPW')
+query_go_a = pathway_query(GOPATHWAY_A)
+query_go_b = pathway_query(GOPATHWAY_B)
+queries = [query_r, query_go_a, query_go_b]
+
+
+# In[7]:
+
+driver = GraphDatabase.driver("bolt://neo4j.het.io")
+full_results_df = pd.DataFrame()
+with driver.session() as session:
+    for query in queries:
+        result = session.run(query)
+        result_df = pd.DataFrame((x.values() for x in result), columns=result.keys())
+        full_results_df = full_results_df.append(result_df, ignore_index=True)
+classifier_genes_df = full_results_df.drop_duplicates().sort_values('gene_symbol').reset_index(drop=True)
+classifier_genes_df['entrez_gene_id'] = classifier_genes_df['entrez_gene_id'].astype('str')
+
+
+# In[8]:
```

Comment:
while I agree it doesn't look pretty to have the genes printed here, it is the one of the two most important aspects a cancer biologist would care about in the classifier (tissue specific performance being the other). They are printed here so someone from that side can quickly scan the genes early on in the classifier script